src/acl/UserData.cc

```diff
@@ -86,56 +86,47 @@ void
 ACLUserData::parse()
 {
     debugs(28, 2, "parsing user list");
-
-    char *t = NULL;
-    if ((t = ConfigParser::strtokFile())) {
-        SBuf s(t);
-        debugs(28, 5, "first token is " << s);
-
-        if (s.cmp("-i",2) == 0) {
-            debugs(28, 5, "Going case-insensitive");
-            flags.case_insensitive = true;
-            // due to how the std::set API work, if we want to change
-            // the comparison function we have to create a new std::set
-            UserDataNames_t newUdn(CaseInsensitveSBufCompare);
-            newUdn.insert(userDataNames.begin(), userDataNames.end());
-            swap(userDataNames,newUdn);
-        } else if (s.cmp("REQUIRED") == 0) {
-            debugs(28, 5, "REQUIRED-type enabled");
-            flags.required = true;
-        } else {
-            if (flags.case_insensitive)
-                s.toLower();
-
-            debugs(28, 6, "Adding user " << s);
-            userDataNames.insert(s);
-        }
+    const auto &parser = ConfigParser::Current();
+    auto userNameOrOption = parser.requiredAclToken("user name or option");
+    debugs(28, 5, "first token is " << userNameOrOption);
+    if (strncmp(userNameOrOption, "-i", 2) == 0) {
+        debugs(28, 5, "Going case-insensitive");
+        flags.case_insensitive = true;
+        // due to how the std::set API work, if we want to change
+        // the comparison function we have to create a new std::set
+        UserDataNames_t newUdn(CaseInsensitveSBufCompare);
+        newUdn.insert(userDataNames.begin(), userDataNames.end());
+        swap(userDataNames, newUdn);
+        userNameOrOption = nullptr;
```

Comment:
To resurrect an old [change request](https://github.com/measurement-factory/squid/pull/125#discussion_r676675007) that GitHub does not let me comment on: Would it be possible to preserve (a lot more of) the official code here instead of converting to PeekAtToken() logic? The changes in this method look out of scope. I wonder if we can avoid them without making anything worse? You said that these changes are necessary because strtokFile() is now protected, but I do not yet understand why we cannot use optionalAclToken() instead, to preserve the original code structure (above and below).

Reply:
I tried to restore the old code to the possible extent.